go/test/endtoend/vreplication/cluster_test.go

```diff
@@ -346,32 +388,46 @@ func NewVitessCluster(t *testing.T, name string, cellNames []string, clusterConf
 	err := topo.ManageTopoDir("mkdir", "/vitess/global")
 	require.NoError(t, err)
 	vc.Topo = topo
-	for _, cellName := range cellNames {
+	for _, cellName := range opts.cells {
 		err := topo.ManageTopoDir("mkdir", "/vitess/"+cellName)
 		require.NoError(t, err)
 	}
 
-	vtctld := cluster.VtctldProcessInstance(vc.ClusterConfig.vtctldPort, vc.ClusterConfig.vtctldGrpcPort,
+	vc.setupVtctld()
+	vc.setupVtctl()
+	vc.setupVtctlClient()
+	vc.setupVtctldClient()
+
+	return vc
+}
+
+func (vc *VitessCluster) setupVtctld() {
```

Comment:
Does this have an non-obvious dependency on `setupVtctl()`, because of the `AddCell` parts?

Reply:
If we remove `setupVtctl()` we will need to add it to `setupVtctld()`  but not otherwise.